source/renderer/app/types/i18nTypes.js

```diff
@@ -4,3 +4,16 @@ export type ReactIntlMessage = {
   defaultMessage: string,
   description: string,
 };
+
+type Record<T, V> = {
+  [T]: V,
+};
+
+type MessageFormatPrimitiveValue = string | number | boolean | null;
+
+export type Intl = {
```

Comment:
Let's use the `intlShape` built in type, so we don't need to keep this updated from the package

Reply:
I notice the current approach to set a type definition to `intl` property is to use `intlShape.isRequired`. The problem with this is `intl` will be loose type resulting in runtime errors.

![Screen Shot 2021-11-09 at 15 50 13](https://user-images.githubusercontent.com/2846918/140986307-c3ad9bee-528a-495b-9b66-c1fcfebf2003.png)

Reply:
@renanvalentin cool, makes sense